### DISCUSSION TITLE
Where can I get more DeclarativeBase magic documents?

### DISCUSSION THREAD

**Discussion by wu-clan** [sqlalchemy/sqlalchemy#8654] [Usage Questions] [answered]
Where can I understand the magic method of Base as much as possible?

for example: (from documentation)

```python
from typing import Optional
from sqlalchemy.orm import DeclarativeBase
from sqlalchemy.orm import Mapped
from sqlalchemy.orm import mapped_column


class Base(DeclarativeBase):
    pass


class User(Base):
    __tablename__ = "user_account"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(String(30))
    fullname: Mapped[Optional[str]]
```
It looks simple enough now, if I want to add comment for each column:

```python
# Now look like
class User(Base):
    __tablename__ = "user_account"

    id: Mapped[int] = mapped_column(primary_key=True, comment='primary key id')
    name: Mapped[str] = mapped_column(String(30), comment='user name')
    fullname: Mapped[Optional[str]] = mapped_colum(comment='user fullname')
```

Can I use magic methods to provide description or more interesting methods in documentation?

**Comment by zzzeek** [comment]
what kind of magic method?   the comment thing right now would be as you have it, using the "comment" field.  if you mean that you'd put `"some comment"` below each entry, we don't have that right now and I'm not sure if we'd want to.

**Comment by wu-clan** [comment, marked as answer]
like this

```python
class User(Base):
    __tablename__ = "user_account"

    id: Mapped[int] = mapped_column(primary_key=True)
    name: Mapped[str] = mapped_column(String(30))
    fullname: Mapped[Optional[str]]
    
    # or other names
    __comment__ = {
        "id": "primary key id",
        ...
    }
```

**Reply by zzzeek** [reply]
I would add an `__init_subclass__` method to your base (https://peps.python.org/pep-0487/) that looks for `__comment__` and then applies it to the attributes you have.

**Reply by wu-clan** [reply]
It looks cool, will try it later, thanks.
